source/validate.h

```diff
@@ -27,30 +27,18 @@
 #ifndef LIBSPIRV_VALIDATE_H_
 #define LIBSPIRV_VALIDATE_H_
 
-#include <algorithm>
-#include <array>
 #include <functional>
-#include <map>
-#include <string>
-#include <unordered_map>
-#include <unordered_set>
 #include <utility>
 #include <vector>
 
-#include "assembly_grammar.h"
-#include "binary.h"
-#include "diagnostic.h"
 #include "instruction.h"
 #include "spirv-tools/libspirv.h"
-#include "spirv_definition.h"
 #include "table.h"
-#include "val/BasicBlock.h"
-
-// Structures
 
 namespace libspirv {
 
 class ValidationState_t;
+class BasicBlock;
```

Comment:
I know forward declarations of classes are common practice.
However, the Google style guide strongly discourages them since they *can* hinder adjusting the API that exports the declared class.
https://google.github.io/styleguide/cppguide.html#Forward_Declarations.

Since BasicBlock is entirely contained in this project, I'll allow this.  In future, let's just #include the defining header.